sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/models/ParallelTransferOptions.java

```diff
@@ -62,7 +63,35 @@ public ParallelTransferOptions(Integer blockSize, Integer numBuffers, ProgressRe
      * {@link BlockBlobAsyncClient#MAX_UPLOAD_BLOB_BYTES}.
      */
     public ParallelTransferOptions(Integer blockSize, Integer numBuffers, ProgressReceiver progressReceiver,
-        Integer maxSingleUploadSize) {
+            Integer maxSingleUploadSize) {
+        this(blockSize, numBuffers, progressReceiver, maxSingleUploadSize, null);
+    }
+
+    /**
+     * Creates a new {@link ParallelTransferOptions} with default parameters applied.
+     *
+     * @param blockSize The block size.
+     * For upload, The block size is the size of each block that will be staged. This value also determines the number
+     * of requests that need to be made. If block size is large, upload will make fewer network calls, but each
+     * individual call will send more data and will therefore take longer. This parameter also determines the size
+     * that each buffer uses when buffering is required and consequently amount of memory consumed by such methods may
+     * be up to blockSize * numBuffers.
+     * @param numBuffers For buffered upload only, the number of buffers is the maximum number of buffers this method
+     * should allocate. Memory will be allocated lazily as needed. Must be at least two. Typically, the larger the
+     * number of buffers, the more parallel, and thus faster, the upload portion  of this operation will be.
+     * The amount of memory consumed by methods using this value may be up to blockSize * numBuffers.
+     * @param progressReceiver {@link ProgressReceiver}
+     * @param maxSingleUploadSize If the size of the data is less than or equal to this value, it will be uploaded in a
+     * single put rather than broken up into chunks. If the data is uploaded in a single shot, the block size will be
+     * ignored. Some constraints to consider are that more requests cost more, but several small or mid-sized requests
+     * may sometimes perform better. In the case of buffered upload, up to this amount of data may be buffered before
+     * any data is sent. Must be greater than 0. May be null to accept default behavior, which is the maximum value the
+     * service accepts for uploading in a single requests and is represented by
+     * {@link BlockBlobAsyncClient#MAX_UPLOAD_BLOB_BYTES}.
+     * @param maxConcurrency The maximum number of parallel requests that will be issued at any given time.
```

Comment:
Does it helps if we add that the min concurrency user can set is 16 or reactor Queues.SMALL_BUFFER_SIZE" https://projectreactor.io/docs/core/3.2.1.RELEASE/api/reactor/util/concurrent/Queues.html#SMALL_BUFFER_SIZE

May be it help customers trying to set a lower value and wonder why it's being ignored.

Reply:
Interesting. I set it to 3, and I saw that fixed the oom error, so I assume it wasn't getting ignored? Let me step through and check

Reply:
As far as I was able to trace the maxConcurrency value, it was still being preserved at 3, so I'm not sure where else this would be getting reset to a higher min.

Reply:
thanks, Rick, I see the min constraint of 16 is applied only when concurrency is set through system property:

```java
public static final int SMALL_BUFFER_SIZE
 = Math.max(16, Integer.parseInt(System.getProperty("reactor.bufferSize.small", "256")));
```

looks like we can explicitly pass a value lower than that.